Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Use the official Golang image to build the Go binary
+FROM golang:1.22 AS builder
+
+# Set the Current Working Directory inside the container
+WORKDIR /app
+
+# Copy go.mod and go.sum files and download dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the source code into the container
+COPY . .
+
+# Build the Go app
+RUN GOOS=linux GOARCH=amd64 go build -o main cmd/productivite/main.go
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding Go build optimization flags</b></div>
<div id="fix">

Consider adding build flags like `-ldflags="-s -w"` to reduce binary size. This can significantly decrease the final image size.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
RUN GOOS=linux GOARCH=amd64 go build -ldflags="-s -w" -o main cmd/productivite/main.go
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/productivite/pull/14#issuecomment-2557461998>#fd823c</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged